packages/payment-processor/src/payment/utils.ts

```diff
@@ -201,6 +219,16 @@ export function validateRequest(
     throw new Error('Both fee address and fee amount have to be declared, or both left empty');
   }
 
+  // Payment network with stream should have both or none of stream flow rate and stream start date
+  const validStreamParams =
+    paymentNetworkId !== ERC777_STREAM || !!expectedFlowRate === !!expectedStartDate;
+
+  if (!validStreamParams) {
+    throw new Error(
+      'Both stream flow rate and stream start date have to be declared, or both left empty',
+    );
+  }
```

Comment:
```suggestion
  if (paymentNetworkId !== ERC777_STREAM) {
    throw new Error(
      'Invalid paymentNetworkId for an ERC777 payment',
    );
  }
  if (!!expectedFlowRate === !!expectedStartDate) {
    throw new Error(
      'Both stream flow rate and stream start date have to be declared, or both left empty',
    );
  }
```

Reply:
This function is called by many payment networks not only by ERC777_STREAM so it should only validate both expectedFlowRate and expectedStartDate are present if the payment network id is ERC777_STREAM. I will fix the code to enforce this requirement.